common/src/main/java/com/wynntils/wc/utils/lootrun/LootrunUtils.java

```diff
@@ -0,0 +1,542 @@
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class LootrunUtils {
+
+    public static final File LOOTRUNS = new File(WynntilsMod.MOD_STORAGE_ROOT, "lootruns");
+
+    private static final List<Integer> COLORS = List.of(
+            ChatFormatting.RED.getColor(),
+            ChatFormatting.GOLD.getColor(),
+            ChatFormatting.YELLOW.getColor(),
+            ChatFormatting.GREEN.getColor(),
+            ChatFormatting.BLUE.getColor(),
+            0x3f00ff,
+            ChatFormatting.DARK_PURPLE.getColor());
+
+    private static final int ACTIVE_COLOR = ChatFormatting.AQUA.getColor();
+    private static final int RECORDING_COLOR = 0xff0000;
+
+    public static LootrunInstance lootrun = null;
+    public static LootrunUncompiled uncompiled = null;
+    public static LootrunInstance recordingCompiled = null;
+    public static LootrunUncompiled recording = null;
+
+    public static RecordingInformation recordingInformation = null;
+
+    public static void register() {
+        WynntilsMod.getEventBus().register(LootrunUtils.class);
+    }
+
+    public static void unregister() {
+        WynntilsMod.getEventBus().register(LootrunUtils.class);
+    }
+
+    @SubscribeEvent
+    public static void render(RenderLevelLastEvent event) {
+        renderLootrun(event, lootrun, ACTIVE_COLOR);
+        renderLootrun(event, recordingCompiled, RECORDING_COLOR);
+    }
+
+    private static void renderLootrun(RenderLevelLastEvent event, LootrunInstance lootrun, int color) {
+        if (lootrun == null) {
+            return;
+        }
+        event.getPoseStack().pushPose();
+        Camera cam = McUtils.mc().gameRenderer.getMainCamera();
+        event.getPoseStack().translate(-cam.getPosition().x, -cam.getPosition().y, -cam.getPosition().z);
+        MultiBufferSource.BufferSource source = McUtils.mc().renderBuffers().bufferSource();
+        var points = lootrun.points();
+        int renderDistance = McUtils.mc().options.renderDistance;
+        BlockPos pos = McUtils.mc().gameRenderer.getMainCamera().getBlockPosition();
+        ChunkPos origin = new ChunkPos(pos);
+        for (int i = 0; i < renderDistance * renderDistance; i++) {
+            int x = i % renderDistance + origin.x - (renderDistance / 2);
+            int z = i / renderDistance + origin.z - (renderDistance / 2);
+            ChunkPos chunk = new ChunkPos(x, z);
+            if (McUtils.mc().level != null && McUtils.mc().level.hasChunk(chunk.x, chunk.z)) {
+                long chunkLong = chunk.toLong();
+                if (points.containsKey(chunkLong)) {
+                    List<List<Point>> locations = points.get(chunkLong);
+                    for (List<Point> locationsInRoute : locations) {
+                        VertexConsumer consumer = source.getBuffer(LootrunRenderType.LOOTRUN_LINE);
```

Comment:
is it really worth creating a whole `RenderType` class just to use it here? the same thing can always be accomplished by calling `Tesselator.getInstance().getBufferBuilder()` and setting the appropriate options. i definitely agree with magicus that this whole function should be split into helpers

Reply:
I think we can keep this, there is not really a harm, and maybe this would be useful when we add the textured version, If any of y'all have a strong opinion against, then let's remove it.